src/components/Budget/BudgetView/BudgetViewContainer.js

```diff
@@ -86,40 +120,52 @@ const BudgetViewContainer = ({ history, match, mutator }) => {
   // eslint-disable-next-line react/prop-types
   const renderActionMenu = ({ onToggle }) => (
     <MenuSection id="budget-actions">
-      <Button
-        buttonStyle="dropdownItem"
-        data-test-edit-budget-button
-        onClick={editBudget}
-      >
-        <Icon
-          size="small"
-          icon="edit"
+      <IfPermission perm="finance.budgets.item.put">
+        <Button
+          buttonStyle="dropdownItem"
+          data-test-edit-budget-button
+          onClick={editBudget}
         >
-          <FormattedMessage id="ui-finance.actions.edit" />
-        </Icon>
-      </Button>
-
-      <Button
-        buttonStyle="dropdownItem"
-        data-test-add-allocation-menu-button
-        onClick={() => {
-          onToggle();
-          toggleAllocateModal();
-        }}
-      >
-        <FormattedMessage id="ui-finance.transaction.allocate" />
-      </Button>
-
-      <Button
-        buttonStyle="dropdownItem"
-        data-test-add-transfer-menu-button
-        onClick={() => {
-          onToggle();
-          toggleTransferModal();
-        }}
-      >
-        <FormattedMessage id="ui-finance.transaction.button.transfer" />
-      </Button>
+          <Icon
+            size="small"
+            icon="edit"
+          >
+            <FormattedMessage id="ui-finance.actions.edit" />
+          </Icon>
+        </Button>
+      </IfPermission>
+
+      <BudgetRemoveAction
```

Comment:
I think it's a common thing to order Delete button last

Reply:
Also, is there a reason to create new component? maybe it makes sense to use the same code style?

Reply:
Fixed, no exceptional reasons)